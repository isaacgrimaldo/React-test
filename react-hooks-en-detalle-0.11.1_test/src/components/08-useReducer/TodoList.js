import React from 'react'
import Proptypes from 'prop-types';

import { TodoListItem } from './TodoListItem'

export const TodoList = ({ todos, handleDelete, handleToggle }) => {
    return (
        <ul className="list-group list-group-flush">
        {
            todos.map( (todo, i) => (
                <TodoListItem 
                    key={ todo.id }
                    todo={ todo }
                    index={ i }
                    handleDelete={ handleDelete }
                    handleToggle={ handleToggle }
                />
            ))
        }
    </ul>
    )
}

TodoList.propTypes = {
    todos: Proptypes.array.isRequired,
    handleDelete: Proptypes.func.isRequired,
    handleToggle: Proptypes.func.isRequired,
}