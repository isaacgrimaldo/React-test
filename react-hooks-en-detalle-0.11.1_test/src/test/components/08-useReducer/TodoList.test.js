import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';

import {todos} from '../../fixtures/todoList'
import {TodoList} from '../../../components/08-useReducer/TodoList';

describe('Pruebas del archivo TodoList' , () => {
    
    const handleDelete = jest.fn(); 
    const handleToggle = jest.fn();
    
    const wrapper = shallow(<TodoList
        todos = {todos} 
        handleDelete = {handleDelete }
        handleToggle = {handleToggle}
     />);

     test('Debe de mostrar correctamente el componente', () => {
            expect(wrapper).toMatchSnapshot();
     });

     test('Debe de mostrar todo los TODOS', () => {
        
           const todoListItems = wrapper.find('TodoListItem').length;
           expect(todoListItems).toBe(todos.length);
     });

     test('Debe pasar correctamente la funciones', () => {
         
        const handleDelete = wrapper.find('TodoListItem').at(0).prop('handleDelete');
        expect(handleDelete).toEqual(expect.any(Function));

        const handleToggle = wrapper.find('TodoListItem').at(1).prop('handleToggle');
        expect(handleToggle).toEqual(expect.any(Function));
     }); 
})