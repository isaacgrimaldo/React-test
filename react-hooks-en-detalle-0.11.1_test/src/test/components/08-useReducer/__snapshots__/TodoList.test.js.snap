// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo TodoList Debe de mostrar correctamente el componente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender Node.js",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender MYSQL",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
